### PR TITLE
jsprocessor for simplewallet

### PR DESCRIPTION
**Pull request by VaralakshmiKA**
jsprocessor for simplewallet
Signed-off-by: test <varalakshmi.ka@intel.com>

### CODE REVIEW

File: jsprocessor/SimpleWalletHandler.js [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    }
+    newBalance = balance - amount
+    console.log("Amount debiting:"+newBalance)
+  }
+  let strNewBalance = newBalance.toString()
+  return _setEntry(context, address, strNewBalance)
+}
+
+//function to make a transfer transaction
+const makeTransfer =(context, senderAddress, amount, recieverAddress)  => (possibleAddressValues) => {
```
Comment (danintel) [inline jsprocessor/SimpleWalletHandler.js]: Receiver and receiver are misspelled throughout this file:

s/eciever/eciever/g

**Review by danintel — CHANGES_REQUESTED** [review]
Spelling error in SimpleWalletHandler.js.

**Review by danintel — APPROVED** [review]

File: jsprocessor/SimpleWalletHandler.js [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    }
+    let amount = update.amount
+    if (amount === null || amount === undefined) {
+      throw new InvalidTransaction('Value is required')
+    }
+    amount = parseInt(amount)
+    if (typeof amount !== "number" ||  amount <= MIN_VALUE) {
+      throw new InvalidTransaction(`Value must be an integer ` + `no less than 1`)
+    }
+        // Select the action to e performed
```
Comment (askmish) [inline jsprocessor/SimpleWalletHandler.js]: Please fix Indentation and spelling errors.

File: jsprocessor/package.json [hunk 3]
```diff
@@ -0,0 +1,42 @@
+{
+  "name": "sawtooth-sdk",
```
Comment (askmish) [inline jsprocessor/package.json]: please update the package.json for jsprocessor.

Comment (askmish) [reply jsprocessor/package.json]: Refer jsclient package.json for example

**Review by askmish — CHANGES_REQUESTED** [review]

**Review by askmish — APPROVED** [review]

File: jsprocessor/SimpleWalletHandler.js [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+    throw new InvalidTransaction("Not enough money to perform transfer operation")
+  }
+  else{
+    console.log("Debiting amount from the sender:"+amount)
+    senderNewBalance = senderBalance -  amount
+    receiverNewBalance = receiverBalance + amount
+    let stateData = senderNewBalance.toString()
+    _setEntry(context, senderAddress, stateData)
+    stateData = receiverNewBalance.toString()
+    console.log("Sender balance:"+senderNewBalance+", Reciever balance:"+receiverNewBalance)
```
Comment (danintel) [inline jsprocessor/SimpleWalletHandler.js:141]: s/Reciever/Receiver/

File: jsprocessor/index.js [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+const { TransactionProcessor } = require('sawtooth-sdk/processor')
+const SimpleWalletHandler = require('./SimpleWalletHandler')
+
+if (process.argv.length < 3) {
+  console.log('missing a validator address')
+  process.exit(1)
+}
+
+const address = process.argv[2]
```
Comment (danintel) [inline jsprocessor/index.js:27]: All the other simplewallet transaction processors have a default for the validator URL.  I suggest something like:

```javascript
const DEFAULT_URL = 'tcp://validator:4004'

// . . .

var address = DEFAULT_URL
if (process.argv.length >= 3) {
  address = process.argv[2]
}
```

**Review by danintel — COMMENTED** [review]
I think the jsprocessor should have the same default validator address as the other transaction processors.  See in-line comment.

**Pull request merged**